src/main/webapp/scripts/tests/perspective-test.js

```diff
@@ -1,4 +1,41 @@
-QUnit.test('perspective - toxicityToPercentString (normal use)', function(assert) {
+const mockPerspectiveResponse = new Response(
+  '{"attributeScores":{"TOXICITY":{"summaryScore":{"value":"0"}}}}',
+  { "status" : 200 }
+);
+
+QUnit.test('getToxicity (no comment input)', function(assert) {
+  const done = assert.async();
+  getDomValue = sinon.fake.returns('');
+  updateDom = sinon.fake();
+  getToxicity().then(() => {
+    assert.equal(updateDom.firstArg, NO_COMMENT_ERROR.message, 'no comment error message text');
+    done();
+  });
+});
+
+QUnit.test('getToxicity (comment length exceeded)', function(assert) {
+  const done = assert.async();
+  getDomValue = sinon.fake.returns('A'.repeat(3001));
+  updateDom = sinon.fake();
+  getToxicity().then(() => {
+    assert.equal(updateDom.firstArg, COMMENT_LENGTH_EXCEEDED_ERROR.message, 'comment length exceeded message text');
+    done();
+  });
+});
+
+QUnit.test('getToxicityFromPerspectiveApi', function(assert) {
+  const done = assert.async();
+  const comment = 'normal comment';
+  sinon.stub(window, 'fetch');
+  window.fetch.returns(Promise.resolve(mockPerspectiveResponse));
+  getToxicityFromPerspectiveApi().then((response) => {
+    assert.equal(response, 0, 'score is 0');
```

Comment:
Nit: Good practice to show in the test that you create mocked response with specific values you are expecting. In this case, it should be something like

```
window.fetch.returns(Promise.resolve(createMockPerspectiveResponse(TOXICITY)));
getToxicityFromPerspectiveApi().then((response) => {
    assert.equal(response, TOXICITY);
```